### PR TITLE
Fix bug that missing eTag when updating azurerm_resource_group_cost_management_export and azurerm_subscription_cost_management_export resource

### PR DESCRIPTION
**Pull request by sinbai**
The purpose of this PR:

> Fix issue [#15010](https://github.com/hashicorp/terraform-provider-azurerm/issues/15010). The issue is caused by missing eTag when updating azurerm_resource_group_cost_management_export and azurerm_subscription_cost_management_export resource in current terraform code. The latest eTag should be set in the update request as described in the [API](https://github.com/Azure/azure-rest-api-specs/blob/1b0a465061c68175898f8f5d27f0301f42ce994c/specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2020-06-01/costmanagement.exports.json#L165). So, I added the eTag to fix this issue.

Test results:

> PASS: TestAccSubscriptionCostManagementExport_basic (266.51s)
> PASS: TestAccSubscriptionCostManagementExport_requiresImport (288.51s)
> PASS: TestAccSubscriptionCostManagementExport_update (463.30s)
> PASS: TestAccResourceGroupCostManagementExport_basic (256.56s)
> PASS: TestAccResourceGroupCostManagementExport_requiresImport (282.96s)
> PASS: TestAccResourceGroupCostManagementExport_update (444.04s)

### CODE REVIEW

File: internal/services/costmanagement/export_base_resource.go [hunk 1]
```diff
@@ -227,7 +227,13 @@ func (br costManagementExportBaseResource) updateFunc() sdk.ResourceFunc {
...
 			}
 
-			if err := createOrUpdateCostManagementExport(ctx, client, metadata, *id); err != nil {
+			// Update operation requires latest eTag to be set in the request.
+			resp, err := client.Get(ctx, id.Scope, id.Name, "")
+			if err != nil {
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if err := createOrUpdateCostManagementExport(ctx, client, metadata, *id, resp.ETag); err != nil {
```
Comment (stephybun) [inline internal/services/costmanagement/export_base_resource.go:236]: The etag could be nil so could we check that beforehand? 
```suggestion
			if resp.ETag == nil {
				return fmt.Errorf("retrieving %s: etag was nil")
			}
                        
			if err := createOrUpdateCostManagementExport(ctx, client, metadata, *id, resp.ETag); err != nil {
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this @sinbai. One minor comment about the etag, once that's resolved this should be good to go!

File: internal/services/costmanagement/export_base_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/costmanagement/export_base_resource.go]: @stephybun Thank you. Fixed.

**Review by stephybun — APPROVED** [review]
Thanks @sinbai, LGTM 🍰

**Pull request merged**